### ISSUE TITLE
Values outside of the specified bounds are tried

### ISSUE THREAD

**Issue by stmax82** [fmfn/BayesianOptimization#6]
Just a small problem I ran into last night while optimizing the max_depth of a sklearn RandomForestClassifier:

I set its bounds to (1, 15) and after a while it stopped with the error:

**ValueError: max_depth must be greater than zero.**

.. which shouldn't happen, since the lower bound is 1.

I was able to reproduce the problem with the following code (python 3.4, numpy 1.10, scipy 0.16, windows x64):

```python
from bayes_opt import BayesianOptimization
import numpy as np

# simulate the random forests accuracy (step function of max_depth + some noise)
def f(x):
    assert x >= 1.0, "%0.27f" % x
    return -int(x) + np.random.randn()*0.2

for i in range(100):
    bo = BayesianOptimization(f=f, pbounds={"x": (1, 15)}, verbose=False)
    bo.maximize(init_points=3, n_iter=250, restarts=1, nugget=0.000001)
```

It takes a few seconds and then fails with **AssertionError: 0.999999999999999888977697537**

I'm not yet sure what really causes the problem, but a simple workaround might be to just set the bounds to (1.01, 15).

PS.: setting the random forests random_state, so it produces exactly the same results every time, does not help.

**Comment by fmfn** [comment]
Oh, I've seen this happening before too. For some reason the optimization of the acquisition function can leads to out of bounds values. It can easily be fixed by simply clipping the outputs of the `acq_max` function. I'll do it later this week. 

For now, I suppose the workaround you mentioned is the way to go.

**Comment by fmfn** [comment]
This issue is fixed in the branch 59a7333 and will be merged soon.
